internal/resources/appplatform/resource.go

```diff
@@ -0,0 +1,620 @@
+package appplatform
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/grafana/authlib/claims"
+	"github.com/grafana/grafana-app-sdk/k8s"
+	sdkresource "github.com/grafana/grafana-app-sdk/resource"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+)
+
+// ResourceModel is a Terraform model for a Grafana resource.
+type ResourceModel struct {
+	Metadata types.Object `tfsdk:"metadata"`
+	Spec     types.Object `tfsdk:"spec"`
+	Options  types.Object `tfsdk:"options"`
+}
+
+// ResourceMetadataModel is a Terraform model for the metadata of a Grafana resource.
+type ResourceMetadataModel struct {
+	UUID      types.String `tfsdk:"uuid"`
+	UID       types.String `tfsdk:"uid"`
+	FolderUID types.String `tfsdk:"folder_uid"`
+	Version   types.String `tfsdk:"version"`
+	URL       types.String `tfsdk:"url"`
+}
+
+// ResourceOptionsModel is a Terraform model for the options of a Grafana resource.
+type ResourceOptionsModel struct {
+	Overwrite types.Bool `tfsdk:"overwrite"`
+}
+
+// ResourceConfig is a configuration for a Grafana resource.
+type ResourceConfig[T sdkresource.Object, L sdkresource.ListObject, S any] struct {
```

Comment:
It seems the generic resources `L` and `S` are not used here 🤔 Why are those needed? Is that something for the future?

Reply:
Also, if that's something needed, might be nice to specify what those types mean in the godoc.

Reply:
Ah, these are leftovers from the previous iteration of the code. Good catch, removed them.